### PR TITLE
Make welcome screen show up on first time run

### LINKED ISSUE #9423: make welcome screen show up on first time run
based on release management discussion, about:welcome should be shown on first time run.

An option is to show it when session is clear and user open browser for the first time, replacing the new tab page one time.

### PR DESCRIPTION
**Pull request by cezaraugusto**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Auditors: @bsclifton
Close #9423

Test Plan:
```
npm run test -- --grep="aboutWelcomeReducer"
npm run unittest -- --grep="tabs API unit tests create"
npm run unittest -- --grep="sessionStore unit tests defaultAppState"
```

Manual test plan:
1. Have your session as-is
2. Open Brave with this PR
3. Open a new tab
4. About welcome shouldn't show
5. Clear session
6. Open Brave
7. New tab should show, followed by about:welcome as the second tab, active
8. Open another tab by any means
9. about:newtab should show alone

Reviewer Checklist:

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

**Comment by bsclifton** [comment]
Moving to 0.18.x

**Comment by bsclifton** [comment]
@cezaraugusto could you rebase please? I'll def review / merge after that 😄  thanks!

**Comment by cezaraugusto** [comment]
ok rebased

**Comment by bsclifton** [comment]
I know this is outside the scope of the pull request, but I wanted to show this:

We'll want to update the website (which is shown in about:welcome) to fix the logo being cropped:
screen shot 2017-06-24 at 10 40 49 pm

File: app/browser/tabs.js [hunk 1]
```diff
@@ -673,6 +678,19 @@ const api = {
       }
       if (!createProperties.url) {
         createProperties.url = newFrameUrl()
+        // don't open welcome screen for general tests
+        if (process.env.NODE_ENV === 'test') {
+          shouldShowWelcomeScreen = false
```
Comment (bsclifton) [inline app/browser/tabs.js]: actually... nevermind, I have a solution for this 😄

**Comment by cezaraugusto** [comment]
@bsclifton awesome thanks for updating, looks great

**Pull request merged**